lib/extensions/rack_session_dalli_patch.rb

```diff
@@ -12,15 +12,26 @@ def delete_sessions(session_ids)
   # cause a load to occur. Consequently, we need to manage things
   # carefully to prevent a deadlock between the Rails Interlock and
   # Dalli's own exclusive lock.
-  def with_lock(*args)
+  def with_block(*args)
     ActiveSupport::Dependencies.interlock.permit_concurrent_loads do
-      super(*args) do
+      super(*args) do |dc|
         ActiveSupport::Dependencies.interlock.running do
-          yield
+          yield dc
         end
       end
     end
   end
 end
 
+# In commit: https://github.com/petergoldstein/dalli/commit/9f9c508afab263a2451f2209c4396daf98d33a1b
+# with_lock was renamed to with_block with a slightly different interface.
+# All versions since 2.7.7 have with_block now.
+# Additionally, we'll detect and warn if the method we're about to prepend/override doesn't exist before we try to prepend it.
+(RackSessionDalliPatch.instance_methods - Module.instance_methods).each do |method|
```

Comment:
I think the following is equivalent

```suggestion
RackSessionDalliPatch.instance_methods(false).each do |method|
```

Reply:
I was hoping to do it dynamically like above, but I had to change it anyway to a hardcoded list since the methods we need in the prepended class don't match the prepended module method names, namely `delete_sessions` calls `delete_session` or `destroy_session` in my latest change.